### PR TITLE
cytoscape: init at 3.4.0

### PR DESCRIPTION
**Pull request by mimame**
This is my first init so I hope that I have done it well.
###### Motivation for this change

Add cytoscape to NixOS
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [X] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/science/misc/cytoscape/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{ stdenv, fetchurl, jre, makeWrapper }:
+
+stdenv.mkDerivation rec {
+  name = "cytoscape-${version}";
+  version = "3.4.0";
+
+  src = fetchurl {
+    url = "http://chianti.ucsd.edu/cytoscape-3.4.0/${name}.tar.gz";
```
Comment (domenkozar) [inline pkgs/applications/science/misc/cytoscape/default.nix]: Another occurance of `${name}` is in the url

**Comment by mimadrid** [comment]
A lot of thanks @domenkozar.

**Comment by mimadrid** [comment]
Sorry for the typo errors, bad spanish worse english. D:

File: pkgs/applications/science/misc/cytoscape/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    sha256 = "065fsqa01w7j85nljwwc0677lfw112xphnyn1c4hb04166q082p2";
+  };
+
+  buildInputs = [jre makeWrapper];
+
+  installPhase = ''
+    mkdir -pv $out/{share,bin}
+    cp -Rv * $out/share/
+
+    cat > $out/bin/cytoscape << EOF
```
Comment (domenkozar) [inline pkgs/applications/science/misc/cytoscape/default.nix]: Shouldn't a symlink be enough? The hashbang will get replaced in final derivation.

**Comment by mimadrid** [comment]
Thanks again @domenkozar, fixed.

File: pkgs/applications/science/misc/cytoscape/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    wrapProgram $out/share/gen_vmoptions.sh \
+      --set JAVA_HOME "${jre}" \
+      --set JAVA  "${jre}/bin/java"
+
+    chmod +x $out/bin/cytoscape
+  '';
+
+  meta = {
+    homepage = "http://www.cytoscape.org";
+    description = "Cytoscape is a general platform for complex network analysis and visualization";
```
Comment (joachifm) [inline pkgs/applications/science/misc/cytoscape/default.nix]: Please remove the package name from the description.

**Comment by joachifm** [comment]
Please squash the commits when you're ready to merge.

**Comment by mimadrid** [comment]
Thanks for all the help @joachifm. 
By the way, I'm packaging other bioinformatic tool but I need some help. How can I do?

**Comment by joachifm** [comment]
@mimadrid you can submit a PR with what you've got so far, doesn't matter if its still work in progress; if you have problems getting started, you can open an issue describing the specific problems you have. (You can try other channels, like IRC or the mailing list, of course).

**Comment by joachifm** [comment]
Pushed to master. Thank you.

**Pull request closed without merging**